File: 202310/515.go
import "math"

/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func largestValues(root *TreeNode) []int {
	var ans []int
	if root == nil {
		return ans
	}
	var q []*TreeNode
	q = append(q, root)

	for len(q) > 0 {
		n := len(q)
		data := math.MinInt32
		for i := 0; i < n; i++ {
			curr := q[0]
			q = q[1:]
			data = max(data, curr.Val)
			if curr.Left != nil {
				q = append(q, curr.Left)
			}
			if curr.Right != nil {
				q = append(q, curr.Right)
			}
		}
		ans = append(ans, data)
	}
	return ans
}

func max(a, b int) int {
	if a > b {
		return a
	}
	return b
}

File: 202412/515.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def largestValues(self, root: Optional[TreeNode]) -> List[int]:
        ans = []
        if root is None:
            return ans
        q = deque()
        q.append(root)
        while q:
            n = len(q)
            ma = float("-inf")
            for _ in range(n):
                p = q.popleft()
                ma = max(ma, p.val)
                if p.left:
                    q.append(p.left)
                if p.right:
                    q.append(p.right)
            ans.append(ma)
        return ans
